crates/re_viewer/src/web_tools.rs

```diff
@@ -37,141 +80,278 @@ pub fn set_url_parameter_and_refresh(key: &str, value: &str) -> Result<(), wasm_
     location.assign(&url.href())
 }
 
-/// Percent-encode the given string so you can put it in a URL.
-pub fn percent_encode(s: &str) -> String {
-    format!("{}", js_sys::encode_uri_component(s))
+static STORED_HISTORY_ENTRY: OnceLock<Arc<Mutex<Option<HistoryEntry>>>> = OnceLock::new();
+
+fn stored_history_entry() -> &'static Arc<Mutex<Option<HistoryEntry>>> {
+    STORED_HISTORY_ENTRY.get_or_init(|| Arc::new(Mutex::new(None)))
```

Comment:
It's good to reduce the scope of things

```suggestion
fn stored_history_entry() -> &'static Arc<Mutex<Option<HistoryEntry>>> {
    static STORED_HISTORY_ENTRY: OnceLock<Arc<Mutex<Option<HistoryEntry>>>> = OnceLock::new();
    STORED_HISTORY_ENTRY.get_or_init(|| Arc::new(Mutex::new(None)))
```